Paintroid/src/androidTest/java/org/catrobat/paintroid/test/espresso/tools/StampToolIntegrationTest.java

```diff
@@ -114,6 +116,52 @@ public void testBorders() {
 		assertEquals(bottomRight, Color.BLACK);
 	}
 
+	@Test
+	public void testStampToolConsidersLayerOpacity() {
+		onToolBarView()
+				.performSelectTool(ToolType.SHAPE);
+
+		onShapeToolOptionsView()
+				.performSelectShape(DrawableShape.RECTANGLE);
+
+		onTopBarView()
+				.performClickCheckmark();
+
+		onToolBarView()
+				.performSelectTool(ToolType.STAMP);
+
+		LayerMenuViewInteraction.onLayerMenuView()
+				.performOpen()
+				.performSetOpacityTo(50, 0)
+				.performClose();
+
+		StampToolViewInteraction.Companion.onStampToolViewInteraction()
+				.performCopy();
+
+		StampTool stampTool = (StampTool) toolReference.getTool();
+		int fiftyPercentOpacityBlack = Color.argb(255 / 2, 0, 0, 0);
+		int centerPixel = stampTool.drawingBitmap.getPixel(stampTool.drawingBitmap.getWidth() / 2, stampTool.drawingBitmap.getHeight() / 2);
+		assertEquals(centerPixel, Color.BLACK);
+
+		StampToolViewInteraction.Companion.onStampToolViewInteraction()
+				.performPaste();
+
+		onDrawingSurfaceView()
+				.checkPixelColor(fiftyPercentOpacityBlack, BitmapLocationProvider.MIDDLE);
+
+		StampToolViewInteraction.Companion.onStampToolViewInteraction()
+				.performCut();
+
+		onDrawingSurfaceView()
+				.checkPixelColor(Color.TRANSPARENT, BitmapLocationProvider.MIDDLE);
+
+		StampToolViewInteraction.Companion.onStampToolViewInteraction()
+				.performPaste();
+
+		onDrawingSurfaceView()
+				.checkPixelColor(fiftyPercentOpacityBlack, BitmapLocationProvider.MIDDLE);
```

Comment:
Pretty much nothing, other than its clearer what the outcome should be, as for assert equals we would first need to get the pixel out out of the bitmap and then assert it instead of having just this line